### ISSUE TITLE
Polishing up the grafting example

### ISSUE THREAD

**Issue by jpfairbanks** [jpfairbanks/SemanticModels.jl#98]
The grafting example really shows off the power of what we are doing here. We take two models and transplant a component from one model onto another model in order to create a third, novel model. Then we use advanced compiler techniques (Cassette) to generate an implementation of that new model that is performant and analyzable. We need to polish that example up to make sure that everything we can automate, we have automated and to make sure that the user API is not "manually munge these Exprs until it works" but instead something that a scientist could plausibly use.

**Comment by jpfairbanks** [comment]
We need to think about what the API interface looks like for the munging of expression trees. I think we should stay close to the `Expr` objects used in Julia because they are the actual grounding for what constitutes `code` in the julia language.

1. finding things
   - variable uses
   - function definitions
   - specific subexpressions
   - constructors and types
2. modifying things
   - variable substitution
   - subexpression substitution
   - optimization transformations
3. combining things
   - plug expression `A` into expression `B` at point `p` 
   - propagating changes downstream of the changes

@mehalter what do you think?

**Comment by mehalter** [comment]
I moved that list to checks, and checked off the functions you added. I think an API like the one you described would be extremely useful and would make it easy enough to make a very clean looking example. Next ups seems to be finding and modifying subexpressions.

Also are we going to handle interchanging variables for subexpressions or functions and vice versa?

**Comment by jpfairbanks** [comment]
Thanks. And yeah I think so we would want to be able to apply the operations in any order.

**Comment by jpfairbanks** [comment]
An example of subexpression substitution:
https://github.com/jpfairbanks/SemanticModels.jl/pull/107/commits/6c23ea48443d3bb6cdc59beb7f93a715a60a8654
```julia
println("Infected individuals recover or die in one step")
# m.transitions[1].args[2].args[4].args[3].args[2].args[2].args[3] = :(:D)
m.transitions[1].args[2].args[3].args[3].args[2].args[2].args[2] = :(:D)

# TODO: make a better api for replacing expressions with other expressions
# rrule′ = replacevar(m.transitions[1].args[2].args[4], :(:S), :(:D))
# TODO: make replacevar handle quotenodes
# rrule′ = replacevar(m.transitions[1].args[2].args[4], :S, :D)
# m.transitions[1].args[2].args[4] = rrule′
```
We need to dig into an expression and find a `QuoteNode` and replace it.

**Comment by jpfairbanks** [comment]
The grafting example has been solved in #129
